src/plugins/dashboard/public/dashboard_app/_dashboard_app_strings.ts

```diff
@@ -139,6 +139,22 @@ export const shareModalStrings = {
       defaultMessage:
         'One or more panels on this dashboard have changed. Before you generate a snapshot, save the dashboard.',
     }),
+  getDraftSharePanelChangesWarning: () =>
+    i18n.translate('dashboard.snapshotShare.panelChangesWarning', {
+      defaultMessage:
+        'You are about to share a dashboard with unsaved changes, and the link may not work properly. Save the dashboard first to create a permanent link.',
+    }),
+  getEmbedSharePanelChangesWarning: () =>
+    i18n.translate('dashboard.embedShare.draftWarning', {
+      defaultMessage:
+        'You are about to create an embedded dashboard with unsaved changes, and the embed code may not work properly. Save the dashboard first to create a permanent embedded dashboard.',
+    }),
+  getDraftShareWarning: (shareType: 'embed' | 'link') =>
+    i18n.translate('dashboard.snapshotShare.draftWarning', {
+      defaultMessage:
+        'This dashboard has unsaved changes. Consider saving your dashboard before generating the {shareType}.',
+      values: { shareType: shareType === 'embed' ? 'embed code' : 'link' },
+    }),
```

Comment:
I like the ternary on `shareType` - can we do the same for `getEmbedSharePanelChangesWarning` + `getDraftSharePanelChangesWarning`?

Reply:
Hey @Heenawter, unlike `getDraftShareWarning` where there's only a different text of the share type,  either `getEmbedSharePanelChangesWarning` or `getDraftSharePanelChangesWarning` don't have a straight forward differentiator.